source/probability_theory_4_finite.Rmd

```diff
@@ -1047,10 +916,75 @@ Convert to a proportion.
 
 Print the results.
 
-Result: kk = .01
+```{python, opts.label="py_ed"}
+N = 10000
+trial_results = np.zeros(N, dtype=bool)
+
+# Deck with 13 spades (ones) and 39 other cards (zeros)
+deck = np.concatenate([
+    np.ones(13),
+    np.zeros(39)
+])
+
+for i in range(N):
+    # Shuffle deck and draw 13 cards
+    hand = np.random.permutation(deck)[:13]
+
+    spades = np.sum(hand)
+
+    if spades >= 5:
+        # Generate a family, zeros for boys, ones for girls
+        children = np.random.randint(0, 2, size=5)
+        girls = np.sum(children)
+        if girls == 4:
+            trial_results[i] = True
+
+k = np.count_nonzero(trial_results)
+
+kk = k / N
+
+print(kk)
+```
+
+Alternative:
+
+```{python, opts.label="py_ed"}
+N = 10000
+trial_spades = np.zeros(N, dtype=int)
+trial_girls = np.zeros(N, dtype=int)
+
+# Deck with 13 spades (ones) and 39 other cards (zeros)
+deck = np.concatenate([
+    np.ones(13),
+    np.zeros(39)
+])
+
+for i in range(N):
+    # Shuffle deck and draw 13 cards
+    hand = np.random.permutation(deck)[:13]
+    spades = np.sum(hand)
+    trial_spades[i] = spades
+
+    # Generate a family, zeros for boys, ones for girls
+    children = np.random.randint(0, 2, size=5)
+    girls = np.sum(children)
+    trial_girls[i] = girls
+
+k = np.count_nonzero((trial_spades >= 5) & (trial_girls == 4))
```

Comment:
Brackets difficult to explain?  `np.logical_and` instead?

Reply:
I think the `&` sign reads more easily here, though.  It may require a sentence or two to explain.  I ran out of steam near the end, as you can tell.

Reply:
Yes - dunno - I go back and forth.  It's a bit annoying to explain the order of operations thing, and it's easy to make very confusing mistakes when using & and forgetting the brackets.

Reply:
In this case it gives an error if you don't have the brackets.